developer/js/lexical-model-compiler/build-trie.ts

```diff
@@ -65,6 +75,58 @@ export function parseWordList(contents: string): WordList {
   return result;
 }
 
+export function compileWordListCharacterSet(wordlist: WordList, searchTermToKey?: (wf: string) => string): string {
+  let charMap: {[char: string]: number} = {};
+  let charList: string[] = [];
+
+  for(let [word, weight] of wordlist) {
+    // There's a problem if we have `null` or `undefined` for a word.
+    if(!word) {
+      throw TypeError();
+    }
+
+    // We want to operate on the search-term form, since that's what is used to look up the words in the Trie.
+    if(searchTermToKey) {
+      word = searchTermToKey(word);
+    }
+
+    for(let i=0; i < word.length; i++) {
+      let code = word.charCodeAt(i);
+      let char = word.charAt(i);
+
+      // Check for surrogate pairs
+      if(code >= 0xD800 && code <= 0xDBFF && word.length > i+1) {
+        let pairedCode = word.charCodeAt(i+1);
+
+        // Ensure it's actually a legit paired code
+        if(pairedCode >= 0xDC00 && pairedCode <= 0xDFFF) {
+          // Computes the actual character's code.
+          //code = (code - 0xD800) * 0x400 + pairedCode - 0xDC00 + 0x10000
```

Comment:
Delete this line. `code` is a dead variable from here on, thus does not need to be reassigned.
```suggestion
```